deeplearning4j/pom.xml

```diff
@@ -92,7 +92,7 @@
         <typesafe.config.version>1.3.0</typesafe.config.version>
         <lombok.version>1.16.20</lombok.version>
         <cleartk.version>2.0.0</cleartk.version>
-        <lucene-solr.version>7.3.0</lucene-solr.version>
+        <lucene-solr.version>7.4.0-SNAPSHOT</lucene-solr.version> <!-- TODO: change to 7.4.0 once it's released -->
```

Comment:
Just to say that the Apache Solr 7.4.0 release voting is in progress currently and I will update and refresh this pull request here once 7.4.0 is released.

Reply:
Opened #5744 separately for clarity to upgrade from 7.3 to 7.4 which requires a `pom.xml` change like here plus two javadoc updates.